keps/sig-windows/3636-windows-csi-host-process-pods/README.md

```diff
@@ -0,0 +1,1414 @@
+  it includes an API group for handling the SMB protocol, which a CSI Driver might not use.
+
+In 1.22, SIG Windows introduced [HostProcess containers](https://kubernetes.io/blog/2021/08/16/windows-hostprocess-containers/)
+as an alternative way to run containers. HostProcess containers run directly in the host
+and behave like to a regular process. The HostProcess containers feature became stable in 1.26.
+
+Using HostProcess containers in CSI Drivers enables CSI Drivers to perform the privileged storage operations
+directly. Most of the drawbacks in the client/server model are no longer present in the new model.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- Define the implementation details to transform CSI Proxy from its client/server model to
+  a go module that can be directly imported by CSI Drivers.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+- Improve the performance of CSI Drivers in Windows - There should be an improvement in the performance by
+  removing the communication aspects between the CSI Driver and CSI Proxy (the protobuf serialization/deserialization,
+  the gRPC call through named pipes). However, this improvement might be negligible, as most of the latency
+  comes from running powershell commands, which is outside the scope of this change.
+- Deprecate the client/server model - This model is still used by the majority of CSI Driver implementations,
+  adopting the new go module model will take time and in the meantime we still plan to maintain
+  both models.
+- Define strict security implementation details - A goal is to understand the security implications of enabling HostProcess
+  containers. We aim to provide guidelines but not implementation details about the components that need to be installed
+  in the cluster.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+This proposal advocates for evolving CSI Proxy from a standalone binary into a Go library that CSI drivers can directly import.
+By leveraging HostProcess containers, CSI drivers for Windows can bundle the necessary privileged functionality,
+eliminating the need for a separate proxy component running on each node.
+This shift simplifies the deployment and maintenance model for both driver developers and cluster administrators.
+
+The core of this proposal is to:
+
+- Refactor the CSI Proxy codebase to expose its API groups as importable Go packages, removing the client/server gRPC architecture.
```

Comment:
In a sig-storage meeting, @bswartz advocated for supporting _both_ solutions (existing CSI-proxy, due to security benefits) and HostProcess mode. As I understand it, the current KEP proposes that HostProcess mode be a replacement for CSI-proxy. So I think this is something worth discussing more, seems like a really important decision that we should close on sooner rather than later.

Right now we maintain the HostProcess implementation in a separate branch over at https://github.com/kubernetes-csi/csi-proxy, which may not be ideal in the world where we decide to maintain both solutions.

cc: @msau42